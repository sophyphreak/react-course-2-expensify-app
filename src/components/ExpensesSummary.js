import React from 'react';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import selectedExpenses from '../selectors/expenses';

import numeral from 'numeral';

export const ExpensesSummary = (props) => (
  <div>
    <p>
      Viewing {props.expenseCount} expense{props.expenseCount !== 1 && 's'} totalling {
        numeral(props.expensesTotal / 100).format('$0,0.00')
      }.
    </p>
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenseCount: selectedExpenses(state.expenses, state.filters).length,
    expensesTotal: selectExpensesTotal(selectedExpenses(state.expenses, state.filters))
  }
};

export default connect(mapStateToProps)(ExpensesSummary);