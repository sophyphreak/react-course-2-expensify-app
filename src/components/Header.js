import React from 'react';
import { NavLink } from 'react-router-dom';
import ExpensesSummary from './ExpensesSummary';

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard </NavLink>
    <NavLink to="/create" activeClassName="is-active">Create expense </NavLink>
    <ExpensesSummary />
  </header>
);

export default Header;