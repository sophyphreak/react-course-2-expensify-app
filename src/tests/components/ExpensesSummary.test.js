import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import selectExpensesTotal from '../../selectors/expenses-total';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary correctly with no expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={expenses[0].amount} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with three expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary 
      expenseCount={expenses.length} 
      expensesTotal={selectExpensesTotal(expenses)} 
    />
  );
  expect(wrapper).toMatchSnapshot();
}); 