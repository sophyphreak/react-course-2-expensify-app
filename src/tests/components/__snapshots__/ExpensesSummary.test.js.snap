// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with no expenses 1`] = `
<div>
  <p>
    Viewing 
    1
     expense
     totalling 
    $1.95
    .
  </p>
</div>
`;

exports[`should render ExpensesSummary correctly with three expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     expense
    s
     totalling 
    $1,141.95
    .
  </p>
</div>
`;
